### PR TITLE
Added reading arguments

### PR DESCRIPTION
**Pull request by kira607**
Now programm can read arguments from command line and check them. Checking unit size based on exeptions. arguments are:
>name_of_programm <file_in_path> <file_out_path> [unit_size]
Unit size is 1 MB by default

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -1,37 +1,74 @@
...
+    {
+        std::cout << colors::red << "Got " << argc << " arguments, but 3 or 4 needed" <<
+        std::endl << "Terminating" << colors::none << std::endl;
+        return 1;
+    }
+    std::string file_in_path = argv[1], file_out_path = argv[2];
+
+    if(file_in_path == file_out_path)
+    {
+        std::cout << colors::red << "AAAA Fuck!" << colors::none << std::endl;
```
Comment (krakacubra) [inline main.cpp]: 1. Please use more clear error messages.
2. Read about the difference btw std::endl and "\n". Decide which one is better to use in this case (and further in the code)

File: main.cpp [hunk 2]
```diff
@@ -1,37 +1,74 @@
...
 
     fin.seekg(0,std::ios::end);
-    int size = fin.tellg();
+    int file_size = fin.tellg();
     fin.seekg(0,std::ios::beg);
 
-    if(test_mode)
+    std::cout << "size of in file: " << file_size << " byte" << std::endl;
+
+    int unit_size;
```
Comment (krakacubra) [inline main.cpp]: unit_size is not initialized

File: main.cpp [hunk 3]
```diff
@@ -1,37 +1,74 @@
...
+    try
+    {
+        std::stringstream convert(
```
Comment (krakacubra) [inline main.cpp]: You commented that convert is initialized by argv[1], but argv[3] used in the code

**Review by krakacubra — DISMISSED** [review]
Fix comments, please.
Write comments in English in the code

File: main.cpp [hunk 4]
```diff
@@ -1,52 +1,104 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <sstream>  //convert
+#include <cstring>  //strcmp
+#include <functional> //hash
 
-bool test_mode = true;
+namespace colors
```
Comment (krakacubra) [inline main.cpp]: better to use enum classes here instead of global variables inside the namespace

File: main.cpp [hunk 5]
```diff
@@ -1,52 +1,104 @@
...
+{
+    char red[]  = "\033[31m";
+    char blue[] = "\033[34m";
+    char none[] = "\033[0m";
+}
+
+bool test_mode = false;
```
Comment (krakacubra) [inline main.cpp]: It is better to use constexpr In such case (hope you do not want to change the mode in the middle of the program execution)

File: main.cpp (hunk 4 above)
Comment (kira607) [reply main.cpp]: I don't think that enum class is the best decision. I'll have to write function that converts enum variable into string, and calling it takes a bit more time than just global variables.

File: CMakeLists.txt [hunk 6]
```diff
@@ -3,4 +3,9 @@ project(untitled)
 
 set(CMAKE_CXX_STANDARD 20)
 
-add_executable(untitled main.cpp)
\ No newline at end of file
+set(CMAKE_CXX_FLAGS "-pthread")
+
+set(SOURCE main.cpp Unit.cpp color.cpp Producer.cpp HashGen.cpp Consumer.cpp)
```
Comment (krakacubra) [inline CMakeLists.txt]: You can use e.g. GLOB here not to forget to e.g. add newly created file
file(GLOB SOURCES *.cpp)
file(GLOB HEADERS *.h)
add_executable(${PROJECT_NAME}
${SOURCES}
${HEADERS}
)

**Review by krakacubra — APPROVED** [review]

**Pull request merged**